vcd/resource_vcd_vapp.go

```diff
@@ -172,94 +168,84 @@ func resourceVcdVAppCreate(d *schema.ResourceData, meta interface{}) error {
 				}
 			}
 
-			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				task, err := vapp.ChangeVMName(d.Get("name").(string))
-				if err != nil {
-					return resource.RetryableError(fmt.Errorf("Error with vm name change: %#v", err))
-				}
+			err = vapp.BlockWhileStatus("UNRESOLVED", vcdClient.MaxRetryTimeout)
+			if err != nil {
+				return fmt.Errorf("error composing vApp: %s", err)
+			}
 
-				return resource.RetryableError(task.WaitTaskCompletion())
-			})
+			task, err := vapp.ChangeVMName(d.Get("name").(string))
+			if err != nil {
+				return fmt.Errorf("error with vm name change: %#v", err)
+			}
+
+			err = task.WaitTaskCompletion()
 			if err != nil {
 				return fmt.Errorf("error changing vmname: %#v", err)
 			}
 
-			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				networks := []map[string]interface{}{map[string]interface{}{
-					"ip":         d.Get("ip").(string),
-					"is_primary": true,
-					"orgnetwork": d.Get("network_name").(string),
-				}}
-				task, err := vapp.ChangeNetworkConfig(networks, d.Get("ip").(string))
-				if err != nil {
-					return resource.RetryableError(fmt.Errorf("error with Networking change: %#v", err))
-				}
-				return resource.RetryableError(task.WaitTaskCompletion())
-			})
+			networks := []map[string]interface{}{map[string]interface{}{
+				"ip":         d.Get("ip").(string),
+				"is_primary": true,
+				"orgnetwork": d.Get("network_name").(string),
+			}}
+			task, err = vapp.ChangeNetworkConfig(networks, d.Get("ip").(string))
+			if err != nil {
+				return fmt.Errorf("error with Networking change: %#v", err)
+			}
+			err = task.WaitTaskCompletion()
 			if err != nil {
 				return fmt.Errorf("error changing network: %#v", err)
 			}
 
 			if ovf, ok := d.GetOk("ovf"); ok {
-				err := retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-					task, err := vapp.SetOvf(convertToStringMap(ovf.(map[string]interface{})))
-
-					if err != nil {
-						return resource.RetryableError(fmt.Errorf("error set ovf: %#v", err))
-					}
-					return resource.RetryableError(task.WaitTaskCompletion())
-				})
+				task, err := vapp.SetOvf(convertToStringMap(ovf.(map[string]interface{})))
+
+				if err != nil {
+					return fmt.Errorf("error set ovf: %#v", err)
```

Comment:
"error set ovf" -> ""error setting the ovf"

Reply:
Fixed